doc/devel/development_setup.rst

```diff
@@ -178,6 +178,9 @@ command ::
 
     python -m pip install --verbose --no-build-isolation --editable ".[dev]"
 
+:::NOTE
+You may need to install pip dependencies pybind11, meson-python, cmake, ninja, numpy for the project to work in editable mode.
```

Comment:
I don't think we should/need to mention dependencies here again. They should be set up in the two previous sections. 

The setup is a multi-step process, which we assume is done in sequence. We also would not write in this section again "make sure that you have cloned the repo".

If anything is missing concerning dependencies, it should be add to https://matplotlib.org/devdocs/devel/development_setup.html#create-a-dedicated-environment or https://matplotlib.org/devdocs/devel/development_setup.html#install-dependencies.

Reply:
@timhoffm Are the dev dependencies enough to setup matplotlib in dev mode? 

`pip install -r requirements/dev/dev-requirements.txt` 

If I remember from our conversation with @story645 it is not very clear on which dependencies to install and that's what we need an explicit note for.

Reply:
Yeah if that's the case then we need to include build deps somewhere or reference transitively. 

Can you reference how is conda doing it? I mean we are sure at this point that the pip dev dependencies don't have it.

Reply:
>make a new ci-requirements file with the contents of dev-requirements 

ok

>find and replace so ci knows to use it

I am confused. Why would we want to mess with the CI builds? The problem was while trying to install it locally in an editable mode.

>add the new build-requirements file to the list in dev-requirements.

Not sure I understand. We change the CI build req. Add it to dev req?

Reply:
We should be clear that the reason the dependencies need to be manually pre-installed is the `--no-build-isolation` not the `-e`.

Reply:
Alright, I have made some fixes and I need one more clarification from @tacaswell 

>We should be clear that the reason the dependencies need to be manually pre-installed is the --no-build-isolation not the -e.

Can you elaborate a bit more or point me to relevant docs for why does that matter? 

Right now I have moved the dependecies to dev ones and I have pointed the CI to those requirements.txt files. 

Do you see this breaking things?

Reply:
The other way as I see being suggested is I separate `meson-python, numpy, pybind11, setuptools-scm` in a build-requirements.txt and then reference it for dev installs?

I am just curious how the current PR breaks/interferes with the setup.

Reply:
Oh wait, I think I get that we don't want all the dev dependencies to be installed in CI run.

Reply:
The following set of commands work now:

```
		python3 -m venv /tmp/clean_build
		. /tmp/clean_build/bin/activate
		pip install ".[dev]"
		pip install -ve . --no-build-isolation
```

Reply:
The CI pipeline is failing [here](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=39506&view=logs&j=b69f81ad-e873-5389-cb38-86053e18a577&t=6b61e752-90a7-583a-d38d-8b6757c5e8cc&l=66) as they are trying to compile pybind11. This is happening because the dev dependencies are being installed in editable mode. 

Why are we installing third party dependencies in editable mode? For local development only matplotlib needs to be in editable mode.

Reply:
But there defnitely is an editable flag being applied somwhere in the CI.